### PR TITLE
[Added]: Cart Page and Add to cart functionality.

### PR DESCRIPTION
**Pull request by munjalSaksham71**
Added Cart Screen and all the implementation like : Add to cart and remove from cart

Note: As the cart products still in state container- so if you refresh the page , the cart products would disappear.

### CODE REVIEW

File: src/components/CartListing/CartListing.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const CartListing = () => {
+  
+  const { cartState: {cart}, cartDispatch } = useCart();
+
+  const [total, setTotal] = useState();
+  useEffect(() => {
+    setTotal(cart.reduce((acc, curr) => {
+      return acc + Number(curr.price) * curr.qty
+    }, 0))
+  })
```
Comment (JantuDeb) [inline src/components/CartListing/CartListing.jsx]: You don't need to do this calculation in `useEffect()`
```
const total = cart.reduce((acc, curr) => acc + Number(curr.price) * curr.qty, 0)
```
The above code will work fine

File: src/context/product-context.js [hunk 2]
```diff
@@ -29,7 +29,6 @@ const ProductProvider = ({ children }) => {
       (async () => {
         try {
             const { data } = await axios.get("/api/products");
-            console.log("Data:", data);
             dispatch({ type: "SUCCESS", payload: data.products });
           } catch (error) {
             dispatch({ type: "ERROR" });
```
Comment (JantuDeb) [inline src/context/product-context.js:34]: You can also add the error to state since, you are dispatching it

**Review by JantuDeb — APPROVED** [review]
Suggested a changes

File: src/components/CartListing/CartListing.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      </div>
+      ))}
+
+      <div className="wrap cart_total_box">
+        <div className="heading2 ml-5 mt-1">Subtotal</div>
+        <div className="heading4 subtotal_amt ml-5 mt-2 mb-2">Rs. {total}</div>
+        <hr></hr>
+        <button className="btn btn-dark-grey btn_checkout m-1">
+          Proceed&nbsp;to&nbsp;checkout
+        </button>
```
Comment (t007rushi) [inline src/components/CartListing/CartListing.jsx:35]: you can use ```react-icons``` instead here

File: src/context/cart-context.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { createContext, useContext, useEffect, useReducer } from 'react'
+
+const CartContext = createContext();
+
+const CartContextProvider = ({children}) => {
+    const cartReducer = (state, action) => {
+        switch (action.type) {
+            case "ADD_TO_CART":
+                return {...state, cart: [...state.cart, {...action.payload, qty:1}]};
+            case "REMOVE_FROM_CART":
```
Comment (t007rushi) [inline src/context/cart-context.js:10]: you can make constant file with diff exports e.g.  ``` export const ADD_TO_CART = "ADD_TO_CART " ``` for reusability and easy imports

**Review by t007rushi — APPROVED** [review]
Liked the UI overall

File: src/context/cart-context.js (hunk 4 above)
Comment (munjalSaksham71) [reply src/context/cart-context.js:10]: Will create a new PR for it, as i have to change the naming from all the files.

File: src/context/product-context.js (hunk 2 above)
Comment (munjalSaksham71) [reply src/context/product-context.js:34]: I do have an error state before.  you can check that on line 14.

File: src/components/CartListing/CartListing.jsx (hunk 3 above)
Comment (munjalSaksham71) [reply src/components/CartListing/CartListing.jsx:35]: here, i have an output of ```Proceed to checkout``` and IG i have to create a button for it and not icon

**Review by t007rushi — APPROVED** [review]
.

**Pull request merged**